### PR TITLE
Fix setting cgroup permission to user/privileged devices

### LINKED ISSUE #21466: Docker in Docker 1.10.x with 1.11.0-rc1 is kinda broken
Running latest master (but I guess it's true for 1.11.0-rc1 too), docker in docker is not working properly. Creating this issue as a follow up of some irc message :angel: (/cc @jfrazelle @thaJeztah :heart_eyes_cat: :angel: )

> 02:18 <•jessfraz> i think in docker master, /sys/fs/cgroups in priviledged containers is RO, like docker-in-docker on master is broken because it cant write
02:38 <•thaJeztah> jessfraz: thx! have you opened an issue to track it?

The daemon starts but it fails to run containers. For example, using libcompose with a simple compose file (that runs `top` in some busybox containers : https://github.com/docker/libcompose/blob/master/integration/assets/multiple-composefiles-default/docker-compose.yml), it breaks.

```
ERRO[0004] Failed Starting another : Error response from daemon: Cannot start container d1e571c7e4ae19332c0cd66ec999d32eb1f4e79bfd49191e5bbad5ce38b8d761: [9] System
 error: write /sys/fs/cgroup/devices/docker/d1e571c7e4ae19332c0cd66ec999d32eb1f4e79bfd49191e5bbad5ce38b8d761/devices.allow: operation not permitted                 ERRO[0004] Failed to start: another : Error response from daemon: Cannot start container d1e571c7e4ae19332c0cd66ec999d32eb1f4e79bfd49191e5bbad5ce38b8d761: [9] Syste
m error: write /sys/fs/cgroup/devices/docker/d1e571c7e4ae19332c0cd66ec999d32eb1f4e79bfd49191e5bbad5ce38b8d761/devices.allow: operation not permitted                ERRO[0005] Failed Starting simple : Error response from daemon: Cannot start container ce4096a06337fc504ed68cf215bc280ce242409e71f92e03efa3731179102c26: [9] System 
error: write /sys/fs/cgroup/devices/docker/ce4096a06337fc504ed68cf215bc280ce242409e71f92e03efa3731179102c26/devices.allow: operation not permitted                  ERRO[0005] Failed to start: simple : Error response from daemon: Cannot start container ce4096a06337fc504ed68cf215bc280ce242409e71f92e03efa3731179102c26: [9] System
 error: write /sys/fs/cgroup/devices/docker/ce4096a06337fc504ed68cf215bc280ce242409e71f92e03efa3731179102c26/devices.allow: operation not permitted                 FATA[0005] Error response from daemon: Cannot start container d1e571c7e4ae19332c0cd66ec999d32eb1f4e79bfd49191e5bbad5ce38b8d761: [9] System error: write /sys/fs/cgro
up/devices/docker/d1e571c7e4ae19332c0cd66ec999d32eb1f4e79bfd49191e5bbad5ce38b8d761/devices.allow: operation not permitted 
```

A full daemon log is available here : https://gist.github.com/vdemeester/d17c414b876abbf58f64.

Docker inside Docker is v1.10.3 and started with `docker daemon --debug --host unix:///go/src/github.com/docker/libcompose/docker.sock --storage-driver vfs --pidfile ./docker.pid --userland-proxy=true`.

---

* Docker version

```
Client:
 Version:      1.11.0-dev
 API version:  1.24
 Go version:   go1.6
 Git commit:   b64627b
 Built:        Thu Mar 24 08:31:20 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.0-dev
 API version:  1.24
 Go version:   go1.6
 Git commit:   b64627b
 Built:        Thu Mar 24 08:31:20 2016
 OS/Arch:      linux/amd64
```

* Docker info

```
Containers: 16
 Running: 1
 Paused: 0
 Stopped: 15
Images: 262
Server Version: 1.11.0-dev
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 4.4.5-1-ARCH
Operating System: Arch Linux
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.56 GiB
Name: hokkaido
ID: YEDF:TDNS:RY6Q:D6RB:OH7S:EPBP:DBZ5:BAVO:HPEH:ALXT:PLP2:FLDM
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): true
 File Descriptors: 22
 Goroutines: 47
 System Time: 2016-03-24T11:15:20.211478419+01:00
 EventsListeners: 0
Username: vdemeester
Registry: https://index.docker.io/v1/
Labels:
 type=laptop
```

* Kernel

```
Linux hokkaido 4.4.5-1-ARCH #1 SMP PREEMPT Thu Mar 10 07:38:19 CET 2016 x86_64 GNU/Linux
```

### PR DESCRIPTION
**Pull request by tonistiigi**
Fixes #21466

@vdemeester @jfrazelle 

Signed-off-by: Tonis Tiigi tonistiigi@gmail.com

### CODE REVIEW

**Comment by jfrazelle** [comment]
is there a way to add an integration test 0:)

**Comment by vdemeester** [comment]
Testing this manually, it looks OK with docker v1.10.3 and v1.9.1 inside the container, and it seems it is fixed (although I have failure with 1.9.1 and the test intgeration suite of libcompose but I think it's something else).

Thanks @tonistiigi

File: daemon/container_operations_unix.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ import (
 
 func u32Ptr(i int64) *uint32     { u := uint32(i); return &u }
 func fmPtr(i int64) *os.FileMode { fm := os.FileMode(i); return &fm }
+func sPtr(s string) *string      { return &s }
```
Comment (calavera) [inline daemon/container_operations_unix.go]: do we really need this function?

**Comment by tonistiigi** [comment]
@vdemeester @jfrazelle I noticed that we already had tests that read cgroups files directly to check the applied settings. I added these new cases like this also.

File: daemon/container_operations_unix.go (hunk 1 above)
Comment (tonistiigi) [inline daemon/container_operations_unix.go]: Removed it. Not sure if it's better.

**Comment by LK4D4** [comment]
LGTM

**Comment by icecrime** [comment]
LGTM

**Comment by thaJeztah** [comment]
all green! merging

**Pull request merged**